flake8_type_checking/checker.py

```diff
@@ -757,27 +868,60 @@ def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
         self.add_import(node)
 
     def visit_ClassDef(self, node: ast.ClassDef) -> ast.ClassDef:
-        """Note down class names."""
-        has_base_classes = node.bases
-        all_base_classes_ignored = all(
-            isinstance(base, ast.Name) and base.id in self.pydantic_enabled_baseclass_passlist for base in node.bases
+        """Create class scope and Note down class names."""
+        for expr in node.decorator_list:
+            self.visit(expr)
```

Comment:
This would've been previously covered by `generic_visit`. But I can't use `generic_visit` here since not all the subnodes in a `ClassDef` have the same scope.

Reply:
Ah interesting, ok makes sense :+1: